sa/saro.go

```diff
@@ -161,6 +168,49 @@ func ReverseName(domain string) string {
 	return strings.Join(labels, ".")
 }
 
+// EncodeIssuedIP prepares an IP address for the issuedNames table by reversing
+// its `in-addr.arpa` or `ip6.arpa` (reverse DNS style) hostname.
+func EncodeIssuedIP(ip net.IP) string {
+	if ip == nil {
+		return ""
+	}
+	reverseDNS, err := dns.ReverseAddr(ip.String())
+	if err != nil {
+		// This is strictly a fallback in case ReverseAddr() fails, which should
+		// be impossible given the output of net.IP.String() in this context.
+		return ReverseName(ip.String())
+	}
+	cutReverseDNS, _ := strings.CutSuffix(reverseDNS, ".")
+	return ReverseName(cutReverseDNS)
+}
+
+// DecodeIssuedName returns the normal (forward) string form of a hostname or IP
+// address that was stored in the issuedNames table.
+func DecodeIssuedName(issuedName string) (string, error) {
```

Comment:
nit: I think these functions should move to model.go. The existence of `Encode...` makes it clear they don't belong in the read-only file :)